lib/acts_as_tree.rb

```diff
@@ -281,6 +288,27 @@ def self_and_siblings
       parent ? parent.children : self.class.roots
     end
 
+    # Returns all the nodes at the same level in the tree as the current node.
+    #
+    #  root1child1.generation [root1child2, root2child1, root2child2]
+    def generation
+      self_and_generation - [self]
+    end
+
+    # Returns a reference to the current node and all the nodes at the same level as it in the tree.
+    #
+    #  root1child1.generation [root1child1, root1child2, root2child1, root2child2]
+    def self_and_generation
+      self.class.select {|node| node.level == self.level }
+    end
+
+    # Returns level (key) for the current nods generation in the generations hash
```

Comment:
I think this comment needs to be updated, for example:

> Returns the level (depth) of the current node

I don't think the fact that the level is also a valid key in the generations hash matters in this context.